### PR TITLE
Properly manage selection when collapsing `expandOnlyOnTwistieClick` tree node

### LINKED ISSUE #140853: File nesting: collapsing with nested file selected selects parent file's next sibling
Issue Type: <b>Bug</b>

Testing 6612ae0f8b5d9c0d86518ebbd153c724a7cbcf73 which landed in today's Insiders.

1. Set `"explorer.experimental.fileNesting.enabled": true`
2. Open VS Code's own source
3. Expand src\tsconfig.json
4. Select first nested file (src\tsconfig.base.json)
5. Collapse nest
6. src\tsec.exemptions.json gets highlighted
7. Expand src\tsconfig.json again and observe two files are now showing highlighted

VS Code version: Code - Insiders 1.64.0-insider (6612ae0f8b5d9c0d86518ebbd153c724a7cbcf73, 2022-01-17T05:16:09.172Z)
OS version: Windows_NT x64 10.0.19044
Restricted Mode: No

### PR DESCRIPTION
**Pull request by JacksonKearl**
This PR fixes #140853

### CODE REVIEW

**Comment by JacksonKearl** [comment]
cc @joaomoreno this doesn't have to be this iteration, what do you think?

**Comment by joaomoreno** [comment]
I think this has nothing to do with `expandOnlyOnTwistieClick`, but a more general tree issue:

Recording 2022-03-25 at 09 27 23

File: src/vs/base/browser/ui/tree/abstractTree.ts [hunk 1]
```diff
@@ -1197,9 +1197,29 @@ class TreeNodeListMouseController<T, TFilterData, TRef> extends MouseController<
...
 			this.tree.setFocus([location]);
+			const oldSelection = this.list.getSelection();
 			model.setCollapsed(location, undefined, recursive);
 
 			if (expandOnlyOnTwistieClick && onTwistie) {
+				if (node.collapsed) {
+					const childCount = node.children.length;
+					// normally super.onViewPointer(e) would set the selection to the clicked element
+					// but here we only want to collapse the element but not overly change the selection
+					// we need to manaully drop all selected elements that got collapsed and shift up
```
Comment (jogo-) [inline src/vs/base/browser/ui/tree/abstractTree.ts:1208]: Typo: man**au**lly  -> man**ua**lly

**Comment by dbaeumer** [comment]
Moving to April.

**Pull request closed without merging**